### PR TITLE
31 parse date of diagnosis

### PR DESCRIPTION
**Pull request by graefea**

### CODE REVIEW

File: src/MC4R/ParseMC4R.py [hunk 1]
```diff
@@ -38,3 +38,38 @@ def parse_year_of_birth(year_of_birth: int) -> str:
         f'but was {year_of_birth}')
     return f'{year_of_birth}-01-01T00:00:00Z'
     
+
+
+
+def parse_date_of_diagnosis(year,month,day):
```
Comment (frehburg) [inline src/MC4R/ParseMC4R.py:44]: Please add data types for year, month and day as e.g. `year: int`. Seems to be int, but please double check. Also missing spaces before `,`

File: src/MC4R/ParseMC4R.py [hunk 2]
```diff
@@ -38,3 +38,38 @@ def parse_year_of_birth(year_of_birth: int) -> str:
...
+    """Parses a patient's date of diagnosis from ERKER to a Phenopackets Age block
+
+    By the Phenopackets documentation Version 2 the onset of a disease i.e. the time of
+    diagnosis can be saved as a TimeElement (Age, Timestamp, TimeInterval)
+
+    Could be: 
+    * "2021-06-02T00:00:00.00Z" for someone diagnosed on June 2nd 2021
+    * "P38Y7M" for someone diagnosed with the Age of 38 years and 7 months
+    * empty if unknown / not stated
```
Comment (frehburg) [inline src/MC4R/ParseMC4R.py:53]: Please say which one we are choosing

File: src/MC4R/ParseMC4R.py [hunk 3]
```diff
@@ -38,3 +38,38 @@ def parse_year_of_birth(year_of_birth: int) -> str:
...
+
+    Example: 
+    parse_date_of_diagnosis(2018-04-21): 
+    >>> "2018-04-21T00:00:00.00Z"
+
+    Link to Phenopackets documentation, where requirement is defined:
+    https://phenopacket-schema.readthedocs.io/en/latest/disease.html 
+
+    :param age_dg: The age of diagnosis of the patient.
+    :type age_dg: str
```
Comment (frehburg) [inline src/MC4R/ParseMC4R.py:63]: This seems to be outdated

**Review by frehburg — CHANGES_REQUESTED** [review]
Bitte einmal kurz drueber schauen

File: src/MC4R/ParseMC4R.py (hunk 1 above)
Comment (graefea) [reply src/MC4R/ParseMC4R.py]: done, it was strings and passes them over to an int now

File: src/MC4R/ParseMC4R.py (hunk 2 above)
Comment (graefea) [reply src/MC4R/ParseMC4R.py]: done

File: src/MC4R/ParseMC4R.py (hunk 3 above)
Comment (graefea) [reply src/MC4R/ParseMC4R.py]: updated accordingly

File: src/MC4R/ParseMC4R.py [hunk 4]
```diff
@@ -41,15 +41,15 @@ def parse_year_of_birth(year_of_birth: int) -> str:
 
 
 
-def parse_date_of_diagnosis(year,month,day):
+def parse_date_of_diagnosis(year: str,month: str,day: str) -> int:
```
Comment (frehburg) [inline src/MC4R/ParseMC4R.py:44]:
```suggestion
def parse_date_of_diagnosis(year: str, month: str, day: str) -> int:
```

File: src/MC4R/ParseMC4R.py [hunk 5]
```diff
@@ -59,8 +59,8 @@ def parse_date_of_diagnosis(year,month,day):
     Link to Phenopackets documentation, where requirement is defined:
     https://phenopacket-schema.readthedocs.io/en/latest/disease.html 
 
-    :param age_dg: The age of diagnosis of the patient.
-    :type age_dg: str
+    :param year,month,day: The date of diagnosis of a patient.
+    :type year,month,day: str
```
Comment (frehburg) [inline src/MC4R/ParseMC4R.py:63]:
```suggestion
    :param year: The year of diagnosis of a patient.
    :type year: str
    :param month: The month of diagnosis of a patient in the given year.
    :type month: str
    :param day: The day of diagnosis of a patient in the given year and month.
    :type day: str
```

**Review by frehburg — CHANGES_REQUESTED** [review]

**Review by frehburg — APPROVED** [review]

**Pull request merged**